File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const temp = {};
    tangerine.forEach((element) => {
        temp[element] = ++temp[element] || 1;
    });

    const arr = Object.values(temp).sort((a, b) => b - a);

    let sum = 0,
        answer = 0;

    for (let num of arr) {
        answer++;
        sum += num;

        if (sum >= k) break;
    }

    return answer;
}

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    map<int, int> m;

    for_each(tangerine.begin(), tangerine.end(), [&](const auto& el)
    {
        m[el]++;
    });

     using pp = pair<int, int>;
    vector<pp> vec(m.begin(), m.end());
    sort(vec.begin(), vec.end(), [](const pp& a, const pp& b) 
    {
        if (a.second == b.second) return a.first < b.first;
        return a.second > b.second;
    });

    for (const auto& num : vec) 
    {
        answer++;
        if (k > num.second) k -= num.second;
        else break;
    }
        
    return answer;
}